lib/child_process.js

```diff
@@ -19,15 +19,20 @@ const ChildProcess = exports.ChildProcess = child_process.ChildProcess;
 exports.fork = function(modulePath /*, args, options*/) {
 
   // Get options and args arguments.
-  var options, args, execArgv;
-  if (Array.isArray(arguments[1])) {
-    args = arguments[1];
-    options = util._extend({}, arguments[2]);
-  } else if (arguments[1] && typeof arguments[1] !== 'object') {
-    throw new TypeError('Incorrect value of args option');
-  } else {
-    args = [];
-    options = util._extend({}, arguments[1]);
+  var execArgv;
+  var options = {};
+  var args = [];
+  var pos = 1;
+  if (Array.isArray(arguments[pos])) {
```

Comment:
Happy to make the change. A couple of questions, though, mostly to make sure I'm understanding things correctly:

* The code I'm replacing accessed `arguments[1]` without any bounds check, so it was already not being optimized. Or was there something else going on there that made that code optimizable?

* This isn't going to be something that's benchmark-able because each `fork()` call is going to spin up a completely separate Node.js instance and any performance improvement in the single call to `fork()` is going to be lost in the overhead of launching the separate instance. Or am I wrong and this may be reasonably benchmark-able perhaps using some trick I'm not thinking of?